server/server.go

```diff
@@ -232,7 +246,26 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		}
 
 		var err error
-		rawGithubClient, err = vcs.NewGithubClient(userConfig.GithubHostname, githubCredentials, ctxLogger, mergeabilityChecker)
+
+		// Extracting internal client creation into its own function to allow for injecting Status Updater
+		internalClient, err := vcs.NewGithubInternalClient(userConfig.GithubHostname, githubCredentials)
+		if err != nil {
+			return nil, err
+		}
+		pullStatusUpdater := &gh_provider.PullStatusUpdater{Client: internalClient}
+		checksStatusUpdater := &gh_provider.ChecksStatusUpdater{Client: internalClient}
+		if userConfig.EnableGithubChecks {
+			ghStatusUpdater = &gh_provider.FeatureAwareStatusUpdater{
+				Pull:             pullStatusUpdater,
+				Check:            checksStatusUpdater,
+				Logger:           ctxLogger,
+				FeatureAllocator: featureAllocator,
+			}
+		} else {
+			ghStatusUpdater = pullStatusUpdater
+		}
```

Comment:
Nish had a comment about not wanting to oveload the `NewGithubClient()` method with the checks stuff. https://github.com/lyft/atlantis/pull/246#discussion_r857912573 That is why we inject the status updater into the github client. Moreover, since `featureAllocator` has a dependency on `githubClient`, using feature flags inside the github client is not possible.  

I basically do the same thing that you have mentioned [here](https://github.com/lyft/atlantis/blob/comment-volume/server/vcs/provider/github/github_status_updater.go) I can simplify the `FeatureAwareStatusUpdater` to use simple functions `updateStatus()` and `updateChecks()` instead of actual structs with implementation: `PullStatusUpdater` and `ChecksStatusUpdater` to make it simpler.